### PR TITLE
suit: migration of SUIT DFU build system to NCS

### PR DESCRIPTION
**Pull request by robertstypa**
Migration of SUIT from  NEXT to NCS.

`sdsc bundle` (ver>=2) need to be downloaded to be able to build configurations listed above if SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY is overwriten by some path, `CONFIG_SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY` points to folder in which bundle is unpacked (nordic-top.suit manifest is available)

SUIT build system can be tested using multicore/hello_world sample:
```
west build -b nrf54h20dk/nrf54h20/cpuapp -p --sysbuild -- -DSB_CONF_FILE=sysbuild/nrf54h20dk_nrf54h20_cpurad.conf -DCONFIG_SUIT_ENVELOPE=y -DCONFIG_SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY=\"/work/nordic-top/\" -DCONFIG_SUIT_MPI_GENERATE=y
```

With signing enabled (example key_private.pem need to be copied into modules/lib/suit-generator/ncs/ directory):
```
west build -b nrf54h20dk/nrf54h20/cpuapp -p --sysbuild -- -DSB_CONF_FILE=sysbuild/nrf54h20dk_nrf54h20_cpurad.conf -DCONFIG_SUIT_ENVELOPE=y -DCONFIG_SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY=\"/work/nordic-top/\" -DCONFIG_SUIT_MPI_GENERATE=y -DCONFIG_SUIT_ENVELOPE_SIGN=y
```

Signing is done by example script defined using SUIT_ENVELOPE_SIGN_SCRIPT

Ref: NCSDK-26629

Signed-off-by: Robert Stypa <robert.stypa@nordicsemi.no>

### CODE REVIEW

**Review by ahasztag — APPROVED** [review]

**Review by adsz-nordic — APPROVED** [review]

**Review by SylwesterKonczyk — APPROVED** [review]

**Review by e-rk — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
@robertstypa pardon my ignorance, as I'm not familiar with SUIT topic, but I received notification that this PR increases 1,6 k of flash on our nRF52840 build. I wonder why it impacts nRF52, if SUIT is not going to be used on this one (I guess)?

**Comment by kkasperczyk-no** [comment]
> @robertstypa pardon my ignorance, as I'm not familiar with SUIT topic, but I received notification that this PR increases 1,6 k of flash on our nRF52840 build. I wonder why it impacts nRF52, if SUIT is not going to be used on this one (I guess)?

Nevermind, I think that the generated report is not correct and it was calculated based on some other changes. I don't think it's possible that this changed made such impact.

**Review by ahasztag — APPROVED** [review]
Looks good :)

File: subsys/suit/provisioning/soc/Kconfig.nrf54h20 [hunk 1]
```diff
@@ -142,7 +142,7 @@ rsource "Kconfig.template.manifest_config"
 
 config SUIT_MPI_RAD_AREA_PATH
 	string "Path to the file with all radio domain MPI configurations"
-	default "suit_mpi_radiocore_merged.hex"
+	default "suit_mpi_radio_merged.hex"
```
Comment (tomchy) [inline subsys/suit/provisioning/soc/Kconfig.nrf54h20:145]: From the nrfutil docs: they propose to use `Network` for the `--core` argument. I think we may keep the `radio` here and get back to this once the command to provision the device will appear.

**Comment by robertstypa** [comment]
> > @robertstypa pardon my ignorance, as I'm not familiar with SUIT topic, but I received notification that this PR increases 1,6 k of flash on our nRF52840 build. I wonder why it impacts nRF52, if SUIT is not going to be used on this one (I guess)?
> 
> Nevermind, I think that the generated report is not correct and it was calculated based on some other changes. I don't think it is possible that this changed made such impact.

This PR is only for nrf54h20 family. Will check the report.

**Comment by kkasperczyk-no** [comment]
> > > @robertstypa pardon my ignorance, as I'm not familiar with SUIT topic, but I received notification that this PR increases 1,6 k of flash on our nRF52840 build. I wonder why it impacts nRF52, if SUIT is not going to be used on this one (I guess)?
> > 
> > 
> > Nevermind, I think that the generated report is not correct and it was calculated based on some other changes. I don't think it is possible that this changed made such impact.
> 
> This PR is only for nrf54h20 family. Will check the report.

As I wrote above, the report probably included several commits for the diff calculation, so it blamed your PR, but actually not your change introduced size increase.

File: cmake/sysbuild/suit.cmake [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+  #if(NOT EXISTS "${SUIT_ROOT_DIRECTORY}")
+    add_custom_target(
+      suit_prepare_output_folder
+      ALL
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${SUIT_ROOT_DIRECTORY}
+      COMMENT
+      "Create DFU output directory"
+    )
+  #endif()
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:235]: are the commented lines needed?

File: cmake/sysbuild/suit.cmake [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+  set(SUIT_OUTPUT_ARTIFACTS_ITEMS)
+  set(SUIT_OUTPUT_ARTIFACTS_TARGETS)
+  set(CORE_ARGS)
+  set(STORAGE_BOOT_ARGS)
+  if (!CONFIG_HW_REVISION_SOC1)
+    set(STORAGE_UPDATE_ARGS)
+  endif()
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:242]: why is this here?

File: cmake/sysbuild/suit.cmake [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+      add_overlay_dts(flash_companion "${overlay}")
+    endif()
+  endforeach()
+
+  # fixme - sysbuild shall be used to build flash_companion
+  add_child_image(
+    NAME flash_companion
+    SOURCE_DIR "${NRF_DIR}/samples/suit/flash_companion"
+    BOARD ${BOARD}
+  )
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:327]: this needs to go, child/parent image should not be used at all from sysbuild

File: cmake/sysbuild/suit.cmake [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+      #     example usage: set_property(TARGET runners_yaml_props_target PROPERTY hex_file ${merged_hex_file})
+       COMMAND ${CMAKE_COMMAND} -E copy ${OUTPUT_HEX_FILE} ${IMAGE_BINARY_DIR}/zephyr/uicr_merged.hex
+    )
+  endforeach()
+endfunction()
+
+# Enable SUIT envelope generation only if DEFAULT_IMAGE has it enabled.
+sysbuild_get(CONFIG_SUIT_ENVELOPE IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE KCONFIG)
+if(CONFIG_SUIT_ENVELOPE)
+    suit_create_package()
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:376]: 2 space indent

File: cmake/sysbuild/suit.cmake [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+# Sign an envelope using SIGN_SCRIPT.
+#
+# Usage:
+#   suit_sign_envelope(<input_file> <output_file>)
+#
+# Parameters:
+#   'input_file' - path to input unsigned envelope
+#   'output_file' - path to output signed envelope
+function(suit_sign_envelope input_file output_file)
+  sysbuild_get(NRF_DIR IMAGE ${DEFAULT_IMAGE} VAR NRF_DIR CACHE)
```
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:108]: Maybe I misunderstood that but there is not access to NRF_DIR at this point, but I what I see there are other variables which could be used for that purpose:

- SYSBUILD_NRF_MODULE_DIR
- ZEPHYR_NRF_MODULE_DIR

I'm not sure which one is better? Probably SYSBUILD_NRF_MODULE_DIR:
```
function(suit_sign_envelope input_file output_file)
  cmake_path(GET SYSBUILD_NRF_MODULE_DIR PARENT_PATH NRF_DIR_PARENT)
```

[the remark this answers is not in the document]
Comment (nordicjm) [reply cmake/sysbuild/suit.cmake:108]: `ZEPHYR_NRF_MODULE_DIR`

File: cmake/sysbuild/suit.cmake [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+#
+# Usage:
+#   suit_set_absolute_or_relative_path(<path> <relative_root> <output_variable>)
+#
+# Parameters:
+#   'path' - path to resolve
+#   'relative_root' - root folder used in case of relative path
+#   'output_variable' - variable to store results
+function(suit_set_absolute_or_relative_path path relative_root output_variable)
+  if(NOT path MATCHES "^/.*" AND NOT path MATCHES "^[a-zA-Z]:/.*")
```
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:93]: I need to analyze that. Do you have any suggestion how to solve that in more elegant way and to cover all cases?

[the remark this answers is not in the document]
Comment (nordicjm) [reply cmake/sysbuild/suit.cmake:93]: https://cmake.org/cmake/help/latest/command/if.html#is-absolute

File: cmake/sysbuild/suit.cmake [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+# Usage:
+#   suit_create_digest(<input_file> <output_file>)
+#
+# Parameters:
+#   'input_file' - input file to calculate digest on
+#   'output_file' - output file to store calculated digest
+function(suit_create_digest input_file output_file)
+  file(SHA256 ${input_file} CHECKSUM_VARIABLE)
+  file(WRITE ${output_file} ${CHECKSUM_VARIABLE})
+endfunction()
```
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:81]: not necessary, function will be removed and digest calculated directly in the `suit_copy_input_template function`

File: cmake/sysbuild/suit.cmake [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+# Parameters:
+#   'input_file' - path to input unsigned envelope
+#   'output_file' - path to output signed envelope
+function(suit_sign_envelope input_file output_file)
+  sysbuild_get(NRF_DIR IMAGE ${DEFAULT_IMAGE} VAR NRF_DIR CACHE)
+  cmake_path(GET NRF_DIR PARENT_PATH NRF_DIR_PARENT)
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT KCONFIG)
+  suit_set_absolute_or_relative_path(${CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT} ${NRF_DIR_PARENT} SIGN_SCRIPT)
+  if (NOT EXISTS ${SIGN_SCRIPT})
+    message(FATAL_ERROR "DFU: ${CONFIG_SUIT_ENVELOPE_SIGN_SCRIPT} does not exist. Corrupted configuration?")
```
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:113]: I do not have an idea on how to solve this in a better way - but poor user experience is the last thing I want in this code. I would appreciate any hints on how to improve this.

File: cmake/sysbuild/suit.cmake (hunk 2 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake]: Removed!

File: cmake/sysbuild/suit.cmake (hunk 3 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake]: leftover from code copied from -next repository, thx for spotting that

File: cmake/sysbuild/suit.cmake (hunk 5 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake]: Fixed!

File: cmake/sysbuild/suit.cmake (hunk 7 above)
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:95]: Done, thanks for the good hint.

File: cmake/sysbuild/suit.cmake (hunk 6 above)
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake]: Done

File: cmake/sysbuild/suit.cmake (hunk 9 above)
[the remark this answers is not in the document]
Comment (nordicjm) [reply cmake/sysbuild/suit.cmake:114]: I would suggest showing a descriptive error here (not using FATAL_ERROR) and then quitting all actions here (if everything is enclosed in a function then you can just `return()` from it), that way it gives them the ability to fix it. Thoughts @tejlmand ?

**Review by tomchy — APPROVED** [review]

File: cmake/sysbuild/suit.cmake (hunk 9 above)
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:114]: I'll try to use `SEND_ERROR`

File: cmake/sysbuild/suit.cmake [hunk 10]
```diff
@@ -0,0 +1,347 @@
...
+#   'output_file' - path to output binary suit envelope
+function(suit_create_envelope input_file output_file)
+  set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE} ${SUIT_GENERATOR_CLI_SCRIPT}
+    create
+    --input-file ${input_file}
+    --output-file ${output_file}
+  )
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_SIGN IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
```
Comment (ahasztag) [inline cmake/sysbuild/suit.cmake:168]: Could we instead add a "boolean" argument to this function stating if we should sign/not sing the envelope and do the extracting of the `CONFIG_SUIT_ENVELOPE_SIGN` outside of this function?

The reason for that is that the secdom/sysctrl build systems would need to use this function and they do not use sysbuild (and do not have knowledge of the `sysbuild_get` function

File: cmake/sysbuild/suit.cmake [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+      # fixme: uicr_merged is overwriten by new content, runners_yaml_props_target could be used to define
+      #     what shall be flashed, but not sure where to set this! Remove --overlap if ready!
+      #     example usage: set_property(TARGET runners_yaml_props_target PROPERTY hex_file ${merged_hex_file})
+       COMMAND ${CMAKE_COMMAND} -E copy ${OUTPUT_HEX_FILE} ${IMAGE_BINARY_DIR}/zephyr/uicr_merged.hex
+    )
+  endforeach()
+endfunction()
+
+# Enable SUIT envelope generation only if DEFAULT_IMAGE has it enabled.
+sysbuild_get(CONFIG_SUIT_ENVELOPE IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE KCONFIG)
```
Comment (ahasztag) [inline cmake/sysbuild/suit.cmake:344]: Couldn't this be moved inside the if below?

File: cmake/sysbuild/suit.cmake [hunk 12]
```diff
@@ -0,0 +1,347 @@
+#
```
Comment (ahasztag) [inline cmake/sysbuild/suit.cmake:1]: Some of the functions from this file are needed by the sysctrl and secdom build systems - which do not use sysbuild and are not aware of the `sysbuild_get` function. Although the two changes I've suggested hep, it does not seem to be clean to include the whole `suit.cmake` file from the sysctrl and secdom build systems.

Maybe we should create a separate file with such SUIT-utility functions, so that this file can be included by the build-systems not using sysbuild. Functions which could be extracted to this file are:
`suit_copy_artifact_to_output_directory`
`suit_render_template`
`suit_create_envelope`

**Review by ahasztag — CHANGES_REQUESTED** [review]

File: cmake/sysbuild/suit.cmake (hunk 11 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:344]: `if` below relays on the info extracted here. Can we address that by moving mentioned three functions outside this file - as you proposed, we could suit-utility file containing:
`suit_copy_artifact_to_output_directory`
`suit_render_template`
`suit_create_envelope`

File: cmake/sysbuild/suit.cmake (hunk 12 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:1]: Good idea, I'll look into that.

File: cmake/sysbuild/suit.cmake (hunk 10 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake]: Done

File: cmake/sysbuild/suit.cmake (hunk 9 above)
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake]: Improved.`SEND_ERROR` is reported instead of `FATAL_ERROR`

File: cmake/sysbuild/suit.cmake (hunk 12 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:1]: Added on the separate [commit](https://github.com/nrfconnect/sdk-nrf/pull/14689/commits/e14c2cb76f3ec2844e619502091dc9154cc7d4c0), please check if this fit sysctr/secdom needs.

File: cmake/sysbuild/suit.cmake [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+include(${CMAKE_CURRENT_LIST_DIR}/suit_utilities.cmake)
+
+set(SUIT_GENERATOR_BUILD_SCRIPT "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/build.py")
+set(SUIT_GENERATOR_CLI_SCRIPT "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/suit_generator/cli.py")
+set(SUIT_OUTPUT_ARTIFACTS_DIRECTORY "DFU")
```
Comment (ahasztag) [inline cmake/sysbuild/suit.cmake:10]: This lines can and should be moved into suit_utilities.cmake, as that file uses them.

Comment (robertstypa) [reply cmake/sysbuild/suit.cmake]: Fixed!

File: cmake/sysbuild/suit.cmake [hunk 14]
```diff
@@ -0,0 +1,283 @@
...
+  )
+  set(SUIT_OUTPUT_ARTIFACTS_ITEMS)
+  set(SUIT_OUTPUT_ARTIFACTS_TARGETS)
+  set(CORE_ARGS)
+  set(STORAGE_BOOT_ARGS)
+
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION KCONFIG)
+  suit_set_absolute_or_relative_path(${CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION} ${PROJECT_BINARY_DIR} INPUT_TEMPLATES_DIRECTORY)
+  sysbuild_get(ENVELOPE_SIGNATURE IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
+  if (NOT DEFINED ENVELOPE_SIGNATURE)
```
Comment (ahasztag) [inline cmake/sysbuild/suit.cmake:170]: `ENVELOPE_SIGNATURE` sounds like this would be the _content_ of the envelope signature, so I would rename this variable to something like `ENVELOPE_SHALL_BE_SIGNED`

Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:170]: Good point, renamed.

**Review by nordicjm — COMMENTED** [review]
Can these changes be squashed? I was about to comment on the second commit saying a change marked as resolved is not resolved, then saw it was fixed 2 commits later?

File: cmake/sysbuild/suit.cmake [hunk 15]
```diff
@@ -0,0 +1,377 @@
...
+function(suit_create_envelope input_file output_file)
+  set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE} ${SUIT_GENERATOR_CLI_SCRIPT}
+    create
+    --input-file ${input_file}
+    --output-file ${output_file}
+  )
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_SIGN IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
+  if (CONFIG_SUIT_ENVELOPE_SIGN)
```
[the remark this answers is not in the document]
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake]: Fixed!

**Comment by robertstypa** [comment]
> Can these changes be squashed? I was about to comment on the second commit saying a change marked as resolved is not resolved, then saw it was fixed 2 commits later?

Done!

File: cmake/sysbuild/suit.cmake [hunk 16]
```diff
@@ -0,0 +1,283 @@
...
+# Setup task to create final and merged artifact.
+#
+# Usage:
+#   suit_setup_merge()
+#
+function(suit_setup_merge)
+  foreach(image ${IMAGES})
+    set(ARTIFACTS_TO_MERGE)
+
+    sysbuild_get(BINARY_DIR IMAGE ${DEFAULT_IMAGE} VAR APPLICATION_BINARY_DIR CACHE)
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:245]: no need to get this for each loop

**Review by nordicjm — COMMENTED** [review]
in `subsys/suit/provisioning/CMakeLists.txt` you could check `extra_post_build_commands` for the parts from zephyr to remove them, would need
```
 depends:
   - hal_nordic
```
in ncs/zephyr/module.yml then you would get the existing values with e.g.
```
get_property(extra_post_build_commands GLOBAL PROPERTY extra_post_build_commands)

# do changes here

set_property(GLOBAL PROPERTY extra_post_build_commands ${extra_post_build_commands})
```

**Review by ahasztag — APPROVED** [review]

**Comment by robertstypa** [comment]
> in `subsys/suit/provisioning/CMakeLists.txt` you could check `extra_post_build_commands` for the parts from zephyr to remove them, would need
> 
> ```
>  depends:
>    - hal_nordic
> ```
> 
> in ncs/zephyr/module.yml then you would get the existing values with e.g.
> 
> ```
> get_property(extra_post_build_commands GLOBAL PROPERTY extra_post_build_commands)
> 
> # do changes here
> 
> set_property(GLOBAL PROPERTY extra_post_build_commands ${extra_post_build_commands})
> ```

Done!. Works properly - building and flashing have been tested with zephyr/basic/blinky with SUIT enabled.

**Comment by robertstypa** [comment]
> > in `subsys/suit/provisioning/CMakeLists.txt` you could check `extra_post_build_commands` for the parts from zephyr to remove them, would need
> > ```
> >  depends:
> >    - hal_nordic
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > in ncs/zephyr/module.yml then you would get the existing values with e.g.
> > ```
> > get_property(extra_post_build_commands GLOBAL PROPERTY extra_post_build_commands)
> > 
> > # do changes here
> > 
> > set_property(GLOBAL PROPERTY extra_post_build_commands ${extra_post_build_commands})
> > ```
> 
> Done!. Works properly - building and flashing have been tested with zephyr/basic/blinky with SUIT enabled.

It seems that CI is in red with latest changes (dependency added to nrf/zephyr/modules and modification of extra_post_build_commands property). I'll revert latest changes and re-run CI to confirm that.

**Review by jnsgsr — APPROVED** [review]

**Comment by carlescufi** [comment]
DNM until @nordicjm or @tejlmand can review.

File: cmake/sysbuild/suit.cmake [hunk 17]
```diff
@@ -0,0 +1,282 @@
...
+    endif()
+    cmake_path(GET source_template_path FILENAME source_filename)
+    set(input_file "${destination_template_directory}/${source_filename}")
+
+    file(SHA256 ${source_template_path} CHECKSUM_DEFAULT)
+    set(DIGEST_STORAGE "${input_file}.digest")
+    file(STRINGS ${DIGEST_STORAGE} CHECKSUM_STORED)
+    if(NOT ${CHECKSUM_DEFAULT} STREQUAL ${CHECKSUM_STORED})
+      message(SEND_ERROR "DFU: Outdated input SUIT template detected - consider update.\n"
+      "Some changes has been done to the SUIT_ENVELOPE_DEFAULT_TEMPLATE which was used to create your ${input_file}.\n"
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:61]: *Some changes have been done

File: cmake/sysbuild/suit.cmake [hunk 18]
```diff
@@ -0,0 +1,282 @@
...
+      list(APPEND ARTIFACTS_TO_MERGE ${IMAGE_BINARY_DIR}/zephyr/uicr.hex)
+    endif()
+
+    set_property(
+      GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+      COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/mergehex.py
+      --overlap replace
+      -o ${OUTPUT_HEX_FILE}
+       ${ARTIFACTS_TO_MERGE}
+      # fixme: uicr_merged is overwritten by new content, runners_yaml_props_target could be used to define
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:270]: Shouldn't there be a change in zephyr/module.yml for a depends on the nrf hal module?

**Review by nordicjm — APPROVED** [review]

File: cmake/sysbuild/suit.cmake (hunk 18 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:270]: Will be added in the separated PR
